docs/Performance/promise.md

```diff
@@ -0,0 +1,195 @@
+
+If you plan to use this with IE11, make sure to provide the [Object.assign Polyfill](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#polyfill).
+
+```js
+/** 
+ * Returns the result of the OData call
+ * @param {object} payload - The payload which equals to the entity
+ * @param {object} parameters - The parameters like Filters added to the OData call
+ * @returns {Promise} Promise object represents result of the OData call
+ */
+function readMyEntity(payload, parameters) {
+  var model = this.getView().getModel();
+  parameters = parameters || {};
+  return model.metadataLoaded()
+    .then(function () {
+      return new Promise(function (resolve, reject) {
+        var key = model.createKey("/MyEntitySet", payload)
+        // prevent success / error callbacks to be overwritten
+        var params = Object.assign({}, parameters,
+          {
+            success: function (data, response) {
+              // prevent accidently change of response data for subsequent calls
+              var dataCopy = Object.assign({}, data);
+              resolve({ data: dataCopy, response: response });
+            },
+            error: function (error) {
+              // additional error handling when needed
+              reject(error);
+            }
+          });
+        model.read(key, params);
+        // update could look like this:
+        // model.update(key, payload, params);
+        // create could look like this:
+        // model.create("/MyEntitySet", payload, params);
+        // delete could look like this:
+        // model.remove(key, params);
+      })
+    })
+}
+
+readMyEntity({ id: 12345 })
+  .then(function (result) {
+    console.log(result.data, result.response);
+  })
+  .catch(function (error) {
+    console.error(error);
+  });
+```
+
+### Chaining multiple OData calls together
+
+Breaks after the first Promise which does not resolve.  
+The finally method is not supported by the Promise Polyfill for IE11.
+
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#chaining>
+
+```js
+myElement.setBusy(true);
+readMyEntity({ id: 12345 })
+  .then(function (result) {
+    console.log(result.data, result.response);
+    return readMyEntity({ id: 67890 })
+  })
+  .then(function (result) {
+    console.log(result.data, result.response);
+    return readMyEntity({ id: 34567 })
+  })
+  .then(function (result) {
+    console.log(result.data, result.response);
+  })
+  .catch(function (error) {
+    console.error(error);
+  })
+  .finally(function () {
+    myElement.setBusy(false);
+  });
+```
+
+### Run multiple OData calls in parallel
+
+#### Promise.all
+
+Breaks with the first Promise that is rejected.
```

Comment:
What do you think about explaining the order of the returned promises? 
https://gist.github.com/vobu/5b2e46f0670571e394f90bc7acf15256  
had to check this myself before believing in it :)

Reply:
added in the latest commit